### ISSUE TITLE
Need some help...Thanks

### ISSUE THREAD

**Issue by Andrea-lyz** [AndroPlus-org/magisk-module-oplus-cn2global#1]
Hello brother, the day after tomorrow I will receive my friend's One Plus 11 from China brought to Europe, brand new and unpacked. I would like to ask you about the principle of triggering this call lock, according to your research, if I have a valid Chinese SIM card (roaming status), insert the phone and use the SIM network directly / or use another rooted and China VPN mounted phone and share it to the One Plus 11 via VPN hotspot to activate it, will I be able to avoid being locked?
I'd like to do my best to avoid this, the OnePlus service center in Europe is in Poland, and while they offer unlocking services(I'm sure it's true. The information provided by China's OnePlus customer service requires a valid purchase certificate), but it's just too far away, and it would take a week for a courier to get back and forth, and if I try to restore factory settings, like a 9008 swipe, would this regional verification be triggered every time?
Maybe Oxygen OS can bypass it, but until now there is no official firmware available and I am not at all sure if the official firmware will be released after I get the phone.Thank you very much for the magisk module, I will try it when I get the phone.

**Comment by AndroPlus-org** [comment]
The region lock happens when you insert SIM card (if MCC is in blacklist), so you can't bypass it with VPN or hotspot.
I found the way to generate unlock code, but currently I don't have OnePlus 11 or newer model right now so can't test.
If it's possible to open
`com.oplus.engineernetwork.register.explock.RegionNetLockDialogExp`
with Activity Launcher or something, we can input unlocking code to unlock region.

**Comment by Andrea-lyz** [comment]
> The region lock happens when you insert SIM card (if MCC is in blacklist), so you can't bypass it with VPN or hotspot.
> I found the way to generate unlock code, but currently I don't have OnePlus 11 or newer model right now so can't test.
> If it's possible to open
> `com.oplus.engineernetwork.register.explock.RegionNetLockDialogExp`
> with Activity Launcher or something, we can input unlocking code to unlock region.

Thanks for the reply, I probably understand.
One small question though, I also mentioned that I have a valid Chinese SIM card, but it is in international roaming status, does the MCC change due to roaming status? I don't know much about this and didn't find any valid information in Google search.
I was told by China's OnePlus customer service that if I successfully activate the phone in China, going abroad to use the call will not trigger the region lock.

**Comment by AndroPlus-org** [comment]
It depends on how OPPO implemented MCC detection.
It looks like some carrier changes MCC when it is roaming so we need to try first...

**Comment by Andrea-lyz** [comment]
> It depends on how OPPO implemented MCC detection.
> It looks like some carrier changes MCC when it is roaming so we need to try first...

hey bro, I got the official firmware of Oxygen OS for OnePlus 11 India region in advance, the version number is 13.0 A.07, could you help to check if there is still region locking related configuration in Oxygen OS?
Link:
https://gauss-componentotacostmanual-in.allawnofs.com/remove-e95820476dfbf1f5f9ca7de73dde1b43/component-ota/23/02/03/947dfadbafdb40c298f8b8c1d40f0792.zip

Thanks!

**Comment by AndroPlus-org** [comment]
Looks like there's no file (at least same location as CN).

**Comment by Andrea-lyz** [comment]
> 看起来没有文件（至少与 CN 相同的位置）。

That's great, I'll try your module tomorrow first (it looks like it should work), since OnePlus became OPPO, the change from ColorOS to Oxygen OS has become more and more difficult.

**Comment by Andrea-lyz** [comment]
> Looks like there's no file (at least same location as CN).

Brother wow, I have already got my OnePlus 11. I rooted and flashed the unrestricted module without internet connection, and then only inserted the Chinese sim card to activate it. It seems half of the success? After inserting the sim card, the area lock pop-up pops up immediately, but I try to make a call, it seems that it is not restricted? Then I inserted a second sim card, a local one, and found that it seemed that the local sim card was restricted from using 5G signals, while the Chinese sim card could search and use 5G signals normally. And of course camera crashes. Do you have any ideas?

![new ducument](https://user-images.githubusercontent.com/52863141/218731537-5135b010-2940-4308-969c-a94aeb626d67.jpg)
![1676374329320](https://user-images.githubusercontent.com/52863141/218731582-727d009b-3287-4795-94ac-5814b6df4300.jpg)
![Screenshot_2023-02-14-12-46-34-90_426a5588c5110cd01d5af17e47adb223](https://user-images.githubusercontent.com/52863141/218731599-d5318114-50cb-47a1-b4cf-635e698df532.jpg)

**Comment by AndroPlus-org** [comment]
I found another props to disable checking, so please test this:
[magisk-module-test.zip](https://github.com/AndroPlus-org/magisk-module-oplus-cn2global/files/10732881/magisk-module-test.zip)
I removed other files related to location history temporarily so camera should work.

**Comment by Andrea-lyz** [comment]
> 我找到了另一个禁用检查的道具，所以请测试这个： [magisk-module-test.zip](https://github.com/AndroPlus-org/magisk-module-oplus-cn2global/files/10732881/magisk-module-test.zip) 我暂时删除了与位置历史相关的其他文件，所以相机应该可以工作。

It does not work. Same problem, inserting the sim card brings up a restriction popup and 5g does not work, even with the 5g switch on, this sim card searches for a 5g signal on my Xiaomi device (I don't have a 5g contract on, but it doesn't affect my search for a 5g signal). oppo has probably done more work for that restriction. Of course the camera did work.

**Comment by Andrea-lyz** [comment]
> I found another props to disable checking, so please test this: [magisk-module-test.zip](https://github.com/AndroPlus-org/magisk-module-oplus-cn2global/files/10732881/magisk-module-test.zip) I removed other files related to location history temporarily so camera should work.

It does not work. Same problem, inserting the sim card brings up a restriction popup and 5g does not work, even with the 5g switch on, this sim card searches for a 5g signal on my Xiaomi device (I don't have a 5g contract on, but it doesn't affect my search for a 5g signal). oppo has probably done more work for that restriction. Of course the camera did work.

**Comment by AndroPlus-org** [comment]
hmm...
Then please check the result of this command in adb shell:
getprop persist.sys.oplus.radio.globalregionlock

I received Ace 2 and it was "1,1" when it's not in locked state.

**Comment by Andrea-lyz** [comment]
> hmm...
> Then please check the result of this command in adb shell:
> getprop persist.sys.oplus.radio.globalregionlock
> 
> I received Ace 2 and it was "1,1" when it's not in locked state.

wait...Ace 2 is unlock?or 1,1?
![Screenshot_2023-02-15-01-41-51-50_9e8df3d0c7c1f50248b6ee043a653d26.jpg](https://user-images.githubusercontent.com/52863141/218896106-81a71821-6061-46a8-8034-772fc6925211.jpg)
I didn't insert the SIM card. If I insert the SIM card, the system will immediately to prompt.

**Comment by AndroPlus-org** [comment]
I'm at Japan so it's not in blacklist of region lock.

It looks like locked state is saved in rpmb partition, so maybe we have to remove saved data.
What is the result of `getprop persist.sys.rpmb.lock`?
![cap_2023-02-16_203850](https://user-images.githubusercontent.com/4242402/219355263-a2a0bcf5-155c-451f-bf0b-4661727355ed.jpg)

**Comment by Andrea-lyz** [comment]
> rompt.

The getprop does not have any output log (module enabled)
![Screenshot_2023-02-16-13-03-16-33_9e8df3d0c7c1f50248b6ee043a653d26 (1)](https://user-images.githubusercontent.com/52863141/219361223-163f1206-b47b-4298-b651-1051441d3883.jpg)

**Comment by AndroPlus-org** [comment]
Please test this after inserting SIM card:
setprop persist.sys.oplus.radio.globalregionlock 0,0
It looks like "1,1" mean "Region check enabled and Locked", "0,0" mean "disabled and unlocked".

Also please check these results when region check popup is shown:
getprop persist.sys.oplus.radio.haslimited
getprop persist.sys.oplus.pnoticetimes

**Comment by Andrea-lyz** [comment]
> Please test this after inserting SIM card: setprop persist.sys.oplus.radio.globalregionlock 0,0 It looks like "1,1" mean "Region check enabled and Locked", "0,0" mean "disabled and unlocked".
> 
> Also please check these results when region check popup is shown: getprop persist.sys.oplus.radio.haslimited getprop persist.sys.oplus.pnoticetimes

It looks like it's still locked. . . I already tried to contact oppo but they kept me waiting for the mail and never contacted me.
![Screenshot_2023-02-16-19-02-57-61_9e8df3d0c7c1f50248b6ee043a653d26](https://user-images.githubusercontent.com/52863141/219451730-e7874ad7-793c-4946-8ac0-19194edd619f.jpg)

**Comment by AndroPlus-org** [comment]
Can you dial
*#3988691#
in phone app with SIM inserted (with my module enabled and disabled)?

Looks like region lock is enabled in Vietnam and it shows unlock code form.
If the form opens, I can generate unlock code from your IMEI 1.
https://www.facebook.com/story.php?story_fbid=pfbid0UDBtdiB4mQJxfbiGhjUn7Tr4xps54kvfq9w8FVqtApBXZh1pB6vdQxrSBwZJT1sul&id=100087398400309&mibextid=Nif5oz

**Comment by Andrea-lyz** [comment]
> Can you dial *#3988691# in phone app with SIM inserted (with my module enabled and disabled)?
> 
> Looks like region lock is enabled in Vietnam and it shows unlock code form. If the form opens, I can generate unlock code from your IMEI 1. https://www.facebook.com/story.php?story_fbid=pfbid0UDBtdiB4mQJxfbiGhjUn7Tr4xps54kvfq9w8FVqtApBXZh1pB6vdQxrSBwZJT1sul&id=100087398400309&mibextid=Nif5oz

Dial *#3988691# will only pop up an unlocked or unlocked toast prompt, and no window for entering the unlock code will appear.
In addition, I had to hand it over to the OnePlus service center. I found that my OnePlus 11, when connected to the 5.0GHz home WiFi first, turned on the mobile hotspot, and the WiFi signal output by the mobile hotspot was always locked at 2.4Ghz, regardless of the mobile hotspot settings. Select the AP frequency band in 2.4 or 5.0. Conversely, if you connect to the 2.4GHz home WiFi first, the mobile hotspot can turn on 5.0GHz.
I asked some friends who have OnePlus 11 and tried some old devices, they all can dual 5.0GHz (WiFi+AP), I tried flash (fastboot), nothing works, I am very worried that it is a hardware problem, and Europe The service center may not perform any testing and repairs on my device. sad.

**Comment by AndroPlus-org** [comment]
hmm, Wi-Fi 5GHz should be the problem of country code.

For the region lock, I added some extra props in v5 so please try this:
https://github.com/AndroPlus-org/magisk-module-oplus-cn2global/releases

Also please test these 3 commands in su shell:
am broadcast -a oplus.intent.action.REGION_NETWORK_LOCK_UI_SHOW --es "RegionNetlockConstant.NETLOCK_STATUS" "0" --es "RegionNetlockConstant.UNLOCK_TYPE" "1" --receiver-permission "oplus.permission.OPLUS_COMPONENT_SAFE"

am broadcast -a oplus.intent.action.REGION_NETWORK_LOCK_STATUS --es "RegionNetlockConstant.NETLOCK_STATUS" "0" --es "RegionNetlockConstant.UNLOCK_TYPE" "1" --receiver-permission "com.oplus.permission.safe.SECURITY"

am broadcast -a com.oplus.action.UNLOCK_NETWORK_SIM1 --es "phoneId" 0

**Comment by Andrea-lyz** [comment]
> 

It didn't work, in *#988#, the region test is still locked in CN
![Screenshot_2023-02-18-16-24-09-59_5d5a8fdec004d8f82e2ce9e098f4b256](https://user-images.githubusercontent.com/52863141/219874066-80b43162-0511-44ac-822c-9a1ad17ce495.jpg)
![Screenshot_2023-02-18-16-24-15-46_5d5a8fdec004d8f82e2ce9e098f4b256](https://user-images.githubusercontent.com/52863141/219874071-04236a31-63b5-4328-8aec-55f12935733c.jpg)
This is the execution result of the three commands
![屏幕截图 2023-02-18 162252](https://user-images.githubusercontent.com/52863141/219874093-82ae1b92-e028-4fb1-9ed2-1fa1df625b9d.png)

**Comment by AndroPlus-org** [comment]
hmm... One malaysian tester reported he could skip region lock with this module on realme GT Neo5, so maybe the lock you have is something different... I have to research more.

**Comment by AndroPlus-org** [comment]
For the Wi-Fi hotspot 5GHz, you can enable it by changing country code to "US" in EngineerMode (open .EngineerModeMain with root in Activity Launcher app)
![Screenshot_2023-02-25-12-09-58-79_40dbc481ca5b738a325e5182fc08a331](https://user-images.githubusercontent.com/4242402/221340195-a5bdfd73-f242-4ca8-8973-bbd20e23025c.jpg)

There's "Global RegionLock" (for OnePlus) and "Region NetLock" (for realme) and looks like only "Region NetLock" can be unlocked with code.
I ordered UK SIM card to lock my phone, hope it locks...

**Comment by Andrea-lyz** [comment]
> For the Wi-Fi hotspot 5GHz, you can enable it by changing country code to "US" in EngineerMode (open .EngineerModeMain with root in Activity Launcher app) ![Screenshot_2023-02-25-12-09-58-79_40dbc481ca5b738a325e5182fc08a331](https://user-images.githubusercontent.com/4242402/221340195-a5bdfd73-f242-4ca8-8973-bbd20e23025c.jpg)
> 
> There's "Global RegionLock" (for OnePlus) and "Region NetLock" (for realme) and looks like only "Region NetLock" can be unlocked with code. I ordered UK SIM card to lock my phone, hope it locks...

I tried your suggestion, unfortunately, when I opened this page for the first time, the country code that was set showed my current country code, I also tried to change to US, but it didn't work. I also tried WiFi hotspot fix channel set, 5GHz 149 effective, it still doesn't work.
![IMG_20230225_151846](https://user-images.githubusercontent.com/52863141/221362624-9ce2107f-16a1-430d-9e74-776d148f1ac7.jpg)
As for your mention of Global RegionLock and buying a UK SIM card to lock the phone, I think it may not be of much use because I found that the MCC code of the roaming state SIM used in the signal abroad is local, for example, I have inserted a Chinese roaming SIM card to activate the One Plus 11, and in *#*#4636#*#* (OnePlus needs to open this page via Activity Launcher) to check the MCC, the MCC of the network is not China MCC, but belongs to the local MCC of the operator providing roaming.

**Comment by Andrea-lyz** [comment]
@AndroPlus-org I have received a response regarding my mobile hotspot issue. OPPO contacted me this morning from China and informed me that they have analyzed the problem. The problem lies on the Qualcomm side, as the WiFi bandwidth in my location is incompatible with the software. This causes the system to automatically downgrade the mobile hotspot to 2.4GHz to avoid network speed interference. It is an underlying software problem that OPPO has attempted to submit to Qualcomm. However, OPPO also informed me that Qualcomm may not fix the issue temporarily.

Despite OPPO's response, I have some doubts and concerns. Firstly, if the problem originates from Qualcomm, why does it not affect the OnePlus 11 sold in Europe? Secondly, I attempted to activate the hotspot separately (without sharing WLAN) using other CN version phones purchased from China and connected them with the OnePlus 11. I then turned on the hotspot of the OnePlus 11, and the problem still persisted. I raised these doubts during the call, but OPPO's customer service was unable to provide an answer.

We are aware that the hardware of OnePlus in different regions is essentially the same. However, I am not certain if this issue is an exception to the rule. Perhaps my guess is incorrect. In relation to this issue that may belong to Qualcomm, I can confirm that the mobile hotspot problem was replicated in a friend's CN version of the OnePlus 11 in the same region.
